internal/storage/image.go

```diff
@@ -93,6 +93,8 @@ type imageService struct {
 	ctx            context.Context
 }
 
+var ImageBeingPulled sync.Map
```

Comment:
I think it could be useful to have a comment on the usage of this

Reply:
I think this should be part of `type Server` since `pullOperationsInProgress` is already there. Talking about `pullOperationsInProgress`.  Why are we not reusing that map?

The `credentials` are also hashed so it may yield false negatives but I think the chances are very low.

Reply:
Yes, let's reuse `pullOperationsInProgress` and maybe change it to a `sync.Map`